state-synchronizer/src/executor_proxy.rs

```diff
@@ -151,4 +166,58 @@ impl ExecutorProxyTrait for ExecutorProxy {
         );
         Ok(waypoint_li.clone())
     }
+
+    async fn publish_on_chain_config_updates(&mut self, events: Vec<ContractEvent>) -> Result<()> {
+        if events.is_empty() {
+            return Ok(());
+        }
+
+        let access_paths = ON_CHAIN_CONFIG_REGISTRY
+            .iter()
+            .map(|config_id| config_id.access_path())
+            .collect();
+        let curr_configs = self
+            .storage_read_client
+            .batch_fetch_config(access_paths)
+            .await;
+
+        let mut new_configs_map = HashMap::new();
+        for (id, configs) in ON_CHAIN_CONFIG_REGISTRY.iter().zip(curr_configs.iter()) {
+            new_configs_map.insert(id, configs);
+        }
+
+        // calculate deltas
+        let mut changed_configs = HashSet::new();
+        let mut new_configs = HashMap::new();
+        for ((id, config), new_config) in self.on_chain_configs.iter().zip(curr_configs) {
+            new_configs.insert(*id, new_config.clone());
+
+            // the byte array is `None` on storage read failure
+            // We should still publish reconfig notification for failed DB read, to indicate that a
+            // reconfig event did happen, regardless of DB failure
```

Comment:
not sure I agree with this, what if the value doesn't change but db read failed? to me the simplest thing to do on db failure is just panic

Reply:
I didn't want to do something like panic because I didn't want one DB read failure in a batch of config reads to cause all the other config reads to fail. I was thinking about this too - but what if the value *did* change when the db read fail? I thought the best-effort thing to do would be to let subscriber know that some kind of reconfiguration did happen, we just don't know how that affected the config, and the subscribed components can do whatever they can with that info